### PR TITLE
Split bootstrap_cluster.rb in two phases

### PR DESCRIPTION
**Pull request by FedericoCeratto**
Split sequence in two and add some waiting to prevent races.
Stop downcasing hostnames.

### CODE REVIEW

**Comment by kiall** [comment]
We don't actually hardcode the usage of bootstrap cluster specifically, if  just name these `01-setup-velum.rb` and `02-setup-velum.rb` and remove the old one, it should pass.

(Though, another change just landed: https://github.com/kubic-project/automation/pull/60 )

**Comment by kiall** [comment]
and .. https://github.com/kubic-project/automation/pull/82

**Comment by FedericoCeratto** [comment]
Updated

File: velum-bootstrap/spec/features/02-setup-velum.rb [hunk 1]
```diff
@@ -19,11 +19,7 @@
...
   end
 
-  scenario "User registers" do
-    with_screenshot(name: :register) do
-      register
-    end
-  end
+  # User creation has already been done in register_and_configure.rb
 
   scenario "User configures the cluster" do
```
Comment (kiall) [inline velum-bootstrap/spec/features/02-setup-velum.rb]: This step was moved to `01-setup-velum.rb` so should be removed from here.. In theory, it's harmless, but, better to not do it twice :)

File: velum-bootstrap/spec/features/02-setup-velum.rb [hunk 2]
```diff
@@ -34,6 +30,10 @@
   scenario "User accepts all minions" do
     visit "/setup/discovery"
 
+    puts ">>> Wait for minions to show up"
+    # FIXME: implement better detection
+    sleep 300
```
Comment (kiall) [inline velum-bootstrap/spec/features/02-setup-velum.rb]: So - I suspect this is related to bare metal taking longer to provision and PXE boot? A few lines down is the check for all minions to be pending, if you add 300 seconds to it's wait time, it'll do the same thing, without a 5 min delay for faster environments.

File: velum-bootstrap/spec/features/02-setup-velum.rb [hunk 3]
```diff
@@ -46,6 +46,11 @@
     end
     puts "<<< accept-all button enabled"
 
+    puts ">>> Wait to prevent query timeout"
```
Comment (kiall) [inline velum-bootstrap/spec/features/02-setup-velum.rb]: I don't quite follow whats happening here - two sections prior to this waits for all the minions to show up, the section prior to this waits for the page to enable the accept all button (which can be a fraction of a second later, hence the wait), and the section after this just clicks it.

Between these steps, nothing else should be happening.. The page should be static/idle etc?

File: velum-bootstrap/spec/features/01-setup-velum.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
+require "spec_helper"
+require 'yaml'
+
+feature "Register user and configure cluster" do
+
+  let(:node_number) { environment["minions"].count { |element| element["role"] != "admin" } }
+  let(:hostnames) { environment["minions"].map { |m| m["fqdn"] if m["role"] != "admin" }.compact }
+  let(:master_minion) { environment["minions"].detect { |m| m["role"] == "master" } }
```
Comment (kiall) [inline velum-bootstrap/spec/features/01-setup-velum.rb]: Nit: None of these 3 get used in this spec, so we should remove them

**Comment by kiall** [comment]
Another nit `02-setup-velum.rb` should be called `02-bootstrap-lcuster.rb` - velum was setup by `01-setup-velum.rb` :)

**Comment by kiall** [comment]
Also needs https://github.com/kubic-project/automation/commit/0b1696f03b74134171ce49589d8af85bd05e3c36 sucked in

File: velum-bootstrap/spec/features/02-setup-velum.rb (hunk 1 above)
Comment (FedericoCeratto) [inline velum-bootstrap/spec/features/02-setup-velum.rb]: Comment added.

File: velum-bootstrap/spec/features/02-setup-velum.rb (hunk 2 above)
Comment (FedericoCeratto) [inline velum-bootstrap/spec/features/02-setup-velum.rb]: Updated

File: velum-bootstrap/spec/features/01-setup-velum.rb (hunk 4 above)
Comment (FedericoCeratto) [inline velum-bootstrap/spec/features/01-setup-velum.rb]: Removed

**Pull request merged**